src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -799,6 +799,32 @@ else if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExer
         }
     }
 
+    /**
+     * GET /programming-exercises/:exerciseId/with-template-and-solution-participation
+     *
+     * @param exerciseId the id of the programmingExercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and the programming exercise with template and solution participation, or with status 404 (Not Found)
+     */
+    @GetMapping(Endpoints.PROGRAMMING_EXERCISE_WITH_TEMPLATE_AND_SOLUTION_PARTICIPATION)
```

Comment:
@krusche We could also use the REST Call which is above this one and adapt the checks to also allow tutors to use it.

Reply:
I guess this is different enough to keep it as a separate endpoint.

Reply:
The one above also includes the latest result which is probably not needed.
I think we can use a new one for now to fix the bugs.